Sources/SwiftDocC/Model/Rendering/RenderNode/RenderMetadata.swift

```diff
@@ -77,6 +94,8 @@ public struct RenderMetadata: VariantContainer {
     /// Authors can use the `@PageImage` metadata directive to provide custom icon and card images for a page.
     public var images: [TopicImage] = []
     
+    public var customMetadata: [CustomMetadataDictionary] = []
```

Comment:
Can we just use a regular dictionary here? It should just be a String to String mapping I think.

```suggestion
    public var customMetadata: [String : String] = [:]
```

Reply:
I did that at first but then everywhere i had to define it instead of having it at a single place like in CustomMetadataDictionary. Also, somehow i associate dictionary to be defined in struct only(coming from swift old school :)). I can update the name of CustomMetadataDictionary(i wasn't too happy with it but thats what came to my mind)

Reply:
This will also represent the data differently though.

Using an array of structs will achieve this kind of JSON:

```
[
   { key: "fruit", value: "orange" },
   { key: "shape", value: "triangle" },
   { key: "plant", value: "tree" }
]
```

while the OpenAPI spec specifies this:

```
{
   "fruit" : "orange",
   "shape" : "triangle",
   "plant" : "tree"
}
```

which will also be simpler for clients to query.

We could do some custom encoding magic to achieve the same layout using the struct, but I think it's likely less error-prone to just using Swift's built-in support for Dictionary.

What do you think?

Reply:
this makes sense.

Reply:
I have updated this. @ethan-kusters please take a look now.